content/en/user-guide/tools/lambda-tools/hot-reloading/index.md

```diff
@@ -226,6 +226,109 @@ To run the example above, run `make watch`. The script is copying the project mo
 along with all dependencies into the `build/hot` folder, which is then mounted into
 LocalStack's Lambda container.
 
+### Hot reloading for TypeScript Lambdas
+
+You can hot-reload your [TypeScript Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-typescript.html). We will check-out a simple example to create a simple `Hello World!` Lambda function using TypeScript.
+
+#### Setting up the Lambda function
+
+Create a new Node.js project with `npm` or an alternative package manager:
+
+{{< command >}}
+$ npm init -y
+{{< / command >}}
+
+Install the the [@types/aws-lambda](https://www.npmjs.com/package/@types/aws-lambda) and [esbuild](https://esbuild.github.io/) packages in your Node.js project:
+
+{{< command >}}
+$ npm install -D @types/aws-lambda esbuild
+{{< / command >}}
+
+Create a new file named `index.ts`. Add the following code to the new file:
+
+```ts
+import { Context, APIGatewayProxyResult, APIGatewayEvent } from 'aws-lambda';
+
+export const handler = async (event: APIGatewayEvent, context: Context): Promise<APIGatewayProxyResult> => {
+  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+  console.log(`Context: ${JSON.stringify(context, null, 2)}`);
+  return {
+      statusCode: 200,
+      body: JSON.stringify({
+          message: 'Hello World!',
+      }),
+   };
+};
+```
+
+Add a build script to your `package.json` file:
+
+```json
+"scripts": {
+    "build": "esbuild index.ts --bundle --minify --sourcemap --platform=node --target=es2020 --outfile=dist/index.js"
```

Comment:
Could we use `--watch` here to avoid manually running `npm run build` everytime we change something? Would be closer to a "real hot reloading" experience.